Automattic/wp-calypso
# Checkout: Move LineItem and NonProductLineItem to wpcom-checkout package

#### Changes proposed in this Pull Request

This PR adjusts the `LineItem` and `NonProductLineItem` components used in checkout so that they have no Calypso (or Redux) dependencies. It then moves these components to the `@automattic/wpcom-checkout` package. 

By moving the components to a package outside Calypso, it becomes possible to use them in non-Calypso contexts like wp-admin. It also guards against regressions because even though the components are already TypeScript, TypeScript errors are ignored by Calypso while the `wpcom-calypso` package will not compile if there are any present.

This is part of https://github.com/Automattic/wp-calypso/issues/52215

This is needed by https://github.com/Automattic/wp-calypso/pull/52890

Requires D67072-code and D67068-code to be merged in order for P2 Plus and Jetpack Search plan subtitles to be displayed correctly but this should be testable without them.

Here's a screenshot of a Jetpack Search plan subtitle before and after this change:

![jetpack-search-checkout-subtitle](https://user-images.githubusercontent.com/2036909/135893540-18dccbf0-2d2b-438c-9fad-5bff2c1562ef.png)

Here's a screenshot of a p2 plan subtitle before this change:

![p2-checkout-subtitle-before](https://user-images.githubusercontent.com/2036909/135893667-bb1d1b9c-de02-448e-81e6-f496bb809ac5.png)

And here's how it looks after this change (a slight formatting difference in how the price is displayed):

![p2-checkout-subtitle](https://user-images.githubusercontent.com/2036909/135893724-137a5bdf-9638-426e-8c3c-79efd2f2a293.png)

#### Testing instructions

- Add a product to the cart and visit checkout.
- Verify that the line items are displayed with no errors.
